absearch/server.py

```diff
@@ -152,7 +181,26 @@ def add_user_to_cohort(**kw):
 def get_cohort_settings(**kw):
     with app._statsd.timer('get_cohort_settings'):
         try:
-            return app.settings.get(**kw)
+            asked_cohort = kw['cohort']
+            res = app.settings.get(**kw)
+            if asked_cohort == 'default':
+                return res
+
+            cohort = res.get('cohort', 'default')
+            locale = kw['locale']
+            territory = kw['territory']
+            asked_cohort = '.'.join([locale, territory, asked_cohort])
+            cohort = '.'.join([locale, territory, cohort])
+
+            if asked_cohort != cohort:
+                # we're getting discarded
+                app._statsd.incr('discarded', tags=[asked_cohort])
+            else:
+                # we're getting back the cohort settings
+                app._statsd.incr('refresh', tags=[asked_cohort])
```

Comment:
Sorry, the old python-dev guy talking here: this one liner is hard to read and makes it easy to do mistakes. I rather be explicit on this one :)